### PR TITLE
feat: Fir 12126 make engine parameter optional

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Allow users to skip engine parameter. Get default engine url in this case.
Added unit and integration tests

### CODE REVIEW

**Comment by ptiurin** [comment]
Lgtm, just wondering if we can use `https://api.app.firebolt.io/core/v1/account/engines:getURLByDatabaseName` ? Looks like it returns a default engine url, without the extra steps.

**Comment by stepansergeevitch** [comment]
> Lgtm, just wondering if we can use `https://api.app.firebolt.io/core/v1/account/engines:getURLByDatabaseName` ? Looks like it returns a default engine url, without the extra steps.

Oh, that actually seems to be much better, didn't see that endpoint. Will use this one

**Review by ptiurin — APPROVED** [review]
lgtm

File: src/firebolt/async_db/connection.py [hunk 1]
```diff
@@ -136,7 +166,15 @@ async def connect_inner(
         # Mypy checks, this should never happen
         assert database is not None
 
-        if engine_name:
+        if not engine_name and not engine_url:
```
Comment (yuryfirebolt) [inline src/firebolt/async_db/connection.py:169]: Change the documentation on L153 to reflect the changes

File: src/firebolt/async_db/connection.py [hunk 2]
```diff
@@ -65,17 +69,43 @@ async def _resolve_engine_url(
...
+            response.raise_for_status()
+            return response.json()["engine_url"]
+        except (
+            JSONDecodeError,
+            RequestError,
+            RuntimeError,
+            HTTPStatusError,
+            KeyError,
+        ) as e:
+            raise InterfaceError(f"Unable to retrieve default engine endpoint: {e}.")
```
Comment (yuryfirebolt) [inline src/firebolt/async_db/connection.py:99]: What if the real error is wrong credentials or expired/incorrect token?

**Review by yuryfirebolt — CHANGES_REQUESTED** [review]

File: src/firebolt/async_db/connection.py (hunk 2 above)
Comment (stepansergeevitch) [reply src/firebolt/async_db/connection.py:99]: Sure, we could also catch `AuthenticationError`

Comment (stepansergeevitch) [reply src/firebolt/async_db/connection.py:99]: On the other side, I think `AuthenticationError` is pretty representative itself, so I think we should just let it propagate further

**Review by yuryfirebolt — APPROVED** [review]

**Pull request merged**